boa/contracts/vyper/vyper_contract.py

```diff
@@ -708,8 +711,10 @@ def get_logs(self, computation=None, include_child_logs=True):
 
     @cached_property
     def event_for(self):
-        m = self.compiler_data.vyper_module_folded._metadata["type"]
-        return {e.event_id: e for e in m.events.values()}
+        event_types = [
```

Comment:
@charles-cooper is this correct?

Reply:
i would have `m = self.compiler_data.global_ctx` (or can call the intermediate variable `module_t`) on a separate line, and probably update to `m.used_events` actually